storage/storage_test.go

```diff
@@ -1361,20 +1361,23 @@ func TestWithEndpoint(t *testing.T) {
 	os.Setenv("STORAGE_EMULATOR_HOST", originalStorageEmulatorHost)
 }
 
-// Create a client using a combination of custom endpoint and
-// STORAGE_EMULATOR_HOST env variable and verify that raw.BasePath,
-// readHost and scheme are not changed by running operations such as upload
+// Create a client using a combination of custom endpoint and STORAGE_EMULATOR_HOST
+// env variable and verify that the client hits the correct endpoint for several
+// different operations performe in sequence.
+// Verifies also that raw.BasePath, readHost and scheme are not changed
+// after running the operations.
 func TestOperationsWithEndpoint(t *testing.T) {
 	originalStorageEmulatorHost := os.Getenv("STORAGE_EMULATOR_HOST")
 	defer os.Setenv("STORAGE_EMULATOR_HOST", originalStorageEmulatorHost)
 
-	var lastRequestEndpoint string
+	gotURL := make(chan string, 1)
+	gotMethod := make(chan string, 1)
 
-	addr, hClient, close := newTestServerAt(func(w http.ResponseWriter, r *http.Request) {
+	hClient, close := newTestServer(func(w http.ResponseWriter, r *http.Request) {
 		io.Copy(ioutil.Discard, r.Body)
 		fmt.Fprintf(w, "{}")
-		req, _ := url.Parse(r.RequestURI)
-		lastRequestEndpoint = req.RawPath
+		gotURL <- r.RequestURI
```

Comment:
Can you have this send the scheme and hostname to be checked as well?

Reply:
added the hostname - not sure how to get the scheme here - the test server is a TLS server so the requests should always be https and it will fail if you try using http